GDJS/Runtime/runtimegame.ts

```diff
@@ -681,54 +594,150 @@ namespace gdjs {
     }
 
     /**
-     * Load all assets, displaying progress in renderer.
+     * Preload a scene assets as soon as possible in background.
+     */
+    prioritizeLoadingOfLayout(layoutName: string) {
+      // Don't await the layout assets to be loaded.
+      this._resourcesLoader.loadLayoutResources(layoutName);
+    }
+
+    /**
+     * @return The progress of assets loading in background for a scene
+     * (between 0 and 1).
+     */
+    getLayoutLoadingProgress(layoutName: string): number {
+      return this._resourcesLoader.getLayoutLoadingProgress(layoutName);
+    }
+
+    /**
+     * @returns true when all the resources of the given layout are loaded
+     * (but maybe not parsed).
+     */
+    isLayoutAssetsLoaded(layoutName: string): boolean {
+      return this._resourcesLoader.isLayoutAssetsLoaded(layoutName);
+    }
+
+    /**
+     * @returns true when all the resources of the given layout are loaded and
+     * parsed.
+     */
+    areLayoutAssetsReady(layoutName: string): boolean {
+      return this._resourcesLoader.areLayoutAssetsReady(layoutName);
+    }
+
+    /**
+     * Load all assets needed to display the 1st layout, displaying progress in
+     * renderer.
      */
     loadAllAssets(
       callback: () => void,
       progressCallback?: (progress: float) => void
     ) {
-      this.loadAllAssetsAsync(progressCallback).then(callback);
+      this.loadFirstAssets(this.getFirstSceneName(), progressCallback).then(
+        callback
+      );
     }
 
     /**
-     * Load all assets, displaying progress in renderer.
+     * Load all assets needed to display the 1st layout, displaying progress in
+     * renderer.
+     *
+     * When a game is hot-reload, this method can be called with the current
+     * layer.
```

Comment:
```suggestion
     * scene.
```

Be careful about layout/layer, this does not looks super professional for someone reading the docs :) 
In the runtime, let's always use scene. "Layout" was a mistake.